src/dswx_sar/pre_processing.py

```diff
@@ -39,29 +42,43 @@ def validate_gtiff(geotiff_path, value_list):
     2. If the mean value equals any value in the provided 'value_list',
        it suggests that all pixels in the file might have the same value.
 
-    Parameters:
+    Parameters
     -----------
     geotiff_path : str
         The file path to the GeoTIFF file to be checked.
     value_list : List[float]
         A list of values against which the mean pixel value of the GeoTIFF
         file is compared.
 
-    Raises:
+    Returns
     -------
-    ValueError:
-        If any of the checks fail, indicating potential issues
+    validation_result: str
+        If any of the checks fail, return a string that indicates potential issues
         with the GeoTIFF file.
-        - Raises an error if some pixels in the file have NaN values.
-        - Raises an error if the mean pixel value matches any value
+        - Return 'nan_value' if some pixels in the file have NaN values.
+        - Return 'invalid_only' if the mean pixel value matches any value
           in 'value_list', suggesting uniform pixel values across the file.
     """
-    image = dswx_sar_util.read_geotiff(geotiff_path)
-    mean_value = np.mean(image)
+    image = dswx_sar_util.read_geotiff(geotiff_path)    
+    mean_value = np.nanmean(image)
+    print(f'checking {geotiff_path}')
```

Comment:
This print is for debugging or logging? Perhaps remove if debug. Otherwise add logging boilerplate.